### PR TITLE
Enable nullable=enable by default for anything except tests

### PR DESCRIPTION
**Pull request by ViktorHofer**
Source generators were the only project type for which the nullable=enable global setting didn't apply. As all source generators already set nullable=enable, I'm trying to set the nullable setting globally for all projects except for tests.

### CODE REVIEW

File: src/libraries/Directory.Build.props [hunk 1]
```diff
@@ -95,8 +95,6 @@
     <!-- Enable documentation file generation by the compiler for all libraries except for vbproj. -->
     <GenerateDocumentationFile Condition="'$(IsSourceProject)' == 'true' and '$(MSBuildProjectExtension)' != '.vbproj'">true</GenerateDocumentationFile>
     <CLSCompliant Condition="'$(CLSCompliant)' == '' and '$(IsTestProject)' != 'true' and '$(IsTestSupportProject)' != 'true'">true</CLSCompliant>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(IsSourceProject)' == 'true' or '$(IsReferenceAssemblyProject)' == 'true'">
     <!-- Nullability is enabled by default for all src and ref projects. -->
```
Comment (eerhardt) [inline src/libraries/Directory.Build.props:98]: Update comment?

File: src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj [hunk 2]
```diff
@@ -2,8 +2,6 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <LangVersion>10.0</LangVersion>
-    <Nullable>enable</Nullable>
-    <ILLinkTrimAssembly>false</ILLinkTrimAssembly>
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj:6]: Are you sure this isn't needed anymore?

**Review by eerhardt — APPROVED** [review]
LGTM! Thanks for this follow-up. I just had one question.

File: src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj:6]: Yes, `ILLinkTrimAssembly` isn't enabled for source generators: https://github.com/dotnet/runtime/blob/c96b44007e34fb43cb9ffb95a390410dc5b9db0d/src/libraries/Directory.Build.targets#L60.

**Pull request merged**